### PR TITLE
Ensure order-agnostic config prior specification

### PR DESCRIPTION
**Pull request by costachris**
## Changes
The behavior of `keys` seems to have changed in julia 1.8, such that calling `collect(keys())` on the prior constraint and mean dicts separately doesn't necessarily result in the same key ordering. This PR ensures `construct_priors` is order-agnostic.

## Checklist before requesting a review / merging
- [x] I have formatted the code using `julia --project=.dev .dev/climaformat.jl .`
- [x] I have updated all test manifests using `julia --project .dev/up_deps.jl .` with Julia 1.8.2.
- [x] If core features are added, I have added appropriate test coverage.
- [x] If new functions and structs are added, they are documented through docstrings.

### CODE REVIEW

File: src/DistributionUtils.jl [hunk 1]
```diff
@@ -47,7 +47,8 @@ function flatten_config_dict(param_dict::Dict{String, T}) where {T}
             push!(values, value)
         end
     end
-    return (u_names, values)
+    sorted_inds = sortperm(u_names)
+    return (u_names[sorted_inds], values[sorted_inds])
```
Comment (ilopezgp) [inline src/DistributionUtils.jl:51]: Just noticed that the docstring is wrong, this function returns a pair of vectors, not a dictionary.

File: src/Pipeline.jl [hunk 2]
```diff
@@ -109,7 +109,7 @@ function init_calibration(config::Dict{Any, Any}; mode::String = "hpc", job_id::
     )
 
     if !isnothing(prior_μ)
-        @assert collect(keys(params)) == collect(keys(prior_μ))
+        @assert keys(params) == keys(prior_μ)
```
Comment (ilopezgp) [inline src/Pipeline.jl:112]: So, how was it possible that this flag was passing? Wouldn't it fail if the order was not the same? Also, why take out the `collect()`?

Comment (costachris) [reply src/Pipeline.jl:112]: `keys` creates a set, so only the entities have to match; `collect` creates a vector, so the order has to match as well. This is the assertion that caught the problem for me originally. I'll double-check where else we are calling `collect(keys()`.

Comment (ilopezgp) [reply src/Pipeline.jl:112]: Ok, I see. I added that assert to check for the order. Is there a way we can keep checking for the order within `src/Pipeline` now? We are using the collect mechanism in a lot of places, and it seems like the behavior can change from version to version, so we should really have some checks there.

Comment (costachris) [reply src/Pipeline.jl]: I implemented a solution that returns the dictionary key/values in a systematic way. The primary goal is to ensure prior mean, constraints, and l2_reg dicts are unpacked  in the same order. Let me know what you think

**Review by ilopezgp — APPROVED** [review]
LGTM

**Pull request merged**